pkg/snapstore/abs_snapstore.go

```diff
@@ -72,16 +81,88 @@ func NewABSSnapStore(config *brtypes.SnapstoreConfig) (*ABSSnapStore, error) {
 	return GetABSSnapstoreFromClient(config.Container, config.Prefix, config.TempDir, config.MaxParallelChunkUploads, &containerURL)
 }
 
-func getCredentials(prefixString string) (storageAccount string, storageKey string, err error) {
-	storageAccount, err = GetEnvVarOrError(prefixString + absStorageAccount)
+func getCredentials(prefixString string) (string, string, error) {
+
+	// TODO: passing credentials through environment variable will be deprecated by "backup-restore v0.18.0"
+	if _, isSet := os.LookupEnv(prefixString + absStorageAccount); isSet {
+		storageAccount, err := GetEnvVarOrError(prefixString + absStorageAccount)
+		if err != nil {
+			return "", "", err
+		}
+		storageKey, err := GetEnvVarOrError(prefixString + absStorageKey)
+		if err != nil {
+			return "", "", err
+		}
+		return storageAccount, storageKey, nil
+	}
+
+	if _, isSet := os.LookupEnv(prefixString + absCredentialJSONFile); isSet {
+		if filename := os.Getenv(prefixString + absCredentialJSONFile); filename != "" {
+			credentials, err := readABSCredentialsJSON(filename)
+			if err != nil {
+				return "", "", fmt.Errorf("error getting credentials using %v file", filename)
+			}
+			return credentials.StorageAccount, credentials.SecretKey, nil
+		}
+	}
+
+	if _, isSet := os.LookupEnv(prefixString + absCredentialFile); isSet {
+		if dir := os.Getenv(prefixString + absCredentialFile); dir != "" {
+			credentials, err := readABSCredentialFiles(dir)
+			if err != nil {
+				return "", "", fmt.Errorf("error getting credentials from %v dir", dir)
+			}
+			return credentials.StorageAccount, credentials.SecretKey, nil
+		}
+	}
+
+	return "", "", fmt.Errorf("unable to get credentials")
+}
+
+func readABSCredentialsJSON(filename string) (*absCredentials, error) {
+	jsonData, err := ioutil.ReadFile(filename)
 	if err != nil {
-		return "", "", err
+		return nil, err
+	}
+
+	return absCredentialsFromJSON(jsonData)
+}
+
+// absCredentialsFromJSON obtains ABS credentials from a JSON value.
+func absCredentialsFromJSON(jsonData []byte) (*absCredentials, error) {
+	absConfig := &absCredentials{}
+	if err := json.Unmarshal(jsonData, absConfig); err != nil {
+		return nil, err
 	}
-	storageKey, err = GetEnvVarOrError(prefixString + absStorageKey)
+
+	return absConfig, nil
+}
+
+func readABSCredentialFiles(dirname string) (*absCredentials, error) {
+	absConfig := &absCredentials{}
+
+	files, err := ioutil.ReadDir(dirname)
 	if err != nil {
-		return "", "", err
+		return nil, err
 	}
-	return
+
+	for _, file := range files {
```

Comment:
Please see https://github.com/gardener/etcd-backup-restore/pull/428#discussion_r804671024, for other providers as well.